test/functional/wallet_taproot.py

```diff
@@ -499,6 +510,20 @@ def run_test(self):
             [True, False],
             lambda k1, k2: (key(k2), [multi_a(1, ([H_POINT] * rnd_pos) + [k1] + ([H_POINT] * (MAX_PUBKEYS_PER_MULTI_A - 1 - rnd_pos)))])
         )
+
+        # Test 999-of-999
+        self.do_test_k_of_n(999, 999)
```

Comment:
Unfortunately this leads to extreme CPU usage and a `sendtoaddress` RPC timeout even on my 2019 MacBook Pro.

I suggest sticking to a lower maximum for now, so we at least get this useful helper function, and leaving a note like:

```
// TODO: fix or work around performance bottleneck in `sendtoaddress`
// in order to test the maximum Taproot allowed 999-of-999.
```

My Macbook can easily handle 1-of-100. The 30 second timeout is hit somewhere between 1-of-750 and 1-of-1000. The CI machines may be slower (or faster, given that it didn't fail - I didn't check its timeout settings).

In term of fixing the bottleneck, you could try the more modern `send` RPC though I doubt that makes a difference. Maybe @achow101 has an idea?

Reply:
I'll lower it to 100 unless @achow101 thinks the `send` RPC approach would work efficiently here.

Reply:
@Sjors just to make sure, did you mean 100-of-100? Or should I leave this out completely and just keep the 1-of-0 and 1-of-1000 tests further down that utilises the same new helper method?
```
        # Test 100-of-100 
        self.do_test_k_of_n(100, 100)
```

Reply:
The CI machines seem to make it through without timeout:
https://github.com/bitcoin/bitcoin/actions/runs/7221218374/job/19675746505?pr=28212#step:7:3594
https://github.com/bitcoin/bitcoin/actions/runs/7221218374/job/19675746796?pr=28212#step:27:330

However, the test runs for 15 minutes on the Win64 job and for 20 minutes on the macOS job, making it the longest running functional test by far.

Reply:
If it still takes that long you probably want to pick a smaller K and/or N.

> did you mean 100-of-100? Or should I leave this out completely and just keep the 1-of-0 and 1-of-1000 tests further down that utilises the same new helper method?

The other two tests only check the failure case. A 1-of-100 would check the success path. 100-of-100 might be fine too, but maybe slower?

Reply:
100-of-100 seems to run okay, less than 16 sec to complete the whole test. However, after the rebase I see the same behavior on my Mac M1 as you're seeing on your 2019 MacBook Pro. Even the 1-of-999 test doesn't pass without an RPC timeout, which it did before! So perhaps there's been some change while the PR has been open that now causes the test to perform slower? 🤔 

1-of-999 gives me:
`'sendall' RPC took longer than 30.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)`

999-of-999 gives me
`'sendtoaddress' RPC took longer than 30.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)`

Reply:
Interesting. Try cleaning your working directory (e.g. `git clean -dfx`) and building again. To try on an older version of master, you can you check out any old commit and then do a `git cherry-pick ...`.

Reply:
> Interesting. Try cleaning your working directory (e.g. `git clean -dfx`) and building again. To try on an older version of master, you can you check out any old commit and then do a `git cherry-pick ...`.

Git bisecting this I found that the tests start timing out with commit 4f473ea515bc77b9138323dab8a741c063d32e8f 

Are you able to verify this on your setup?

@darosior would you be able to check as well? 🙏

This is how I did it:
```
git checkout 4f473ea515bc77b9138323dab8a741c063d32e8f
git cherry-pick 387c12e0813a863bc9728777719acbafe7b12a34
make
python3 test/functional/wallet_taproot.py
...
test_framework.authproxy.JSONRPCException: 'sendtoaddress' RPC took longer than 30.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)
```
Test previous commit:
```
git checkout HEAD~1
git cherry-pick 387c12e0813a863bc9728777719acbafe7b12a34
make
python3 test/functional/wallet_taproot.py
...
2023-12-15T20:46:16.689000Z TestFramework (INFO): Tests successful
```

Reply:
Great find! Let's track this in a separate issue: #29098